src/rust/src/lib.rs

```diff
@@ -25,6 +47,35 @@ fn is_fips_enabled() -> bool {
     cryptography_openssl::fips::is_enabled()
 }
 
+#[cfg(CRYPTOGRAPHY_OPENSSL_300_OR_GREATER)]
+fn _initialize_legacy_provider() -> CryptographyResult<LoadedProviders> {
+    /* As of OpenSSL 3.0.0 we must register a legacy cipher provider
+    to get RC2 (needed for junk asymmetric private key
+    serialization), RC4, Blowfish, IDEA, SEED, etc. These things
+    are ugly legacy, but we aren't going to get rid of them
+    any time soon. */
+    let load_legacy = env::var("CRYPTOGRAPHY_OPENSSL_NO_LEGACY")
+        .map(|v| v.is_empty() || v == "0")
+        .unwrap_or(true);
+    if load_legacy {
+        let legacy = provider::Provider::load(None, "legacy");
+        _legacy_provider_error(legacy.is_ok())?;
+        let default = provider::Provider::load(None, "default")?;
+        Ok(LoadedProviders::new(Some(default), Some(legacy?)))
+    } else {
+        Ok(LoadedProviders::new(None, None))
+    }
```

Comment:
This does not match the previous behavior, which was to always load the the default provider.